elixir.go

```diff
@@ -138,8 +141,14 @@ func validateToken(rawJwt, jwksURL string) (*jwt.Token, error) {
 
 	// If error is for signature validation
 	if err != nil && v.Errors == jwt.ValidationErrorSignatureInvalid {
-		return nil, fmt.Errorf("signature not valid: %s", err.Error())
+		return nil, "", fmt.Errorf("signature not valid: %s", err.Error())
 	}
 
-	return token, err
+	d, ok := token.Claims.(jwt.MapClaims)["exp"].(float64)
+	if !ok {
+		log.Error("failed to read expiration date from token")
+	}
+	expireDate := time.Unix(int64(d), 0)
```

Comment:
```suggestion
	if !ok {
		return nil, "", fmt.Errorf("failed to read expiration date from token")
	}
	expireDate := time.Unix(int64(d), 0)
```

A valid token must have an expiration date, if we can't read that we need to throw an error.

Reply:
The signature validation that was in place already did an implicit date validation. However, I agree with the point and I adedd some logic to validate dates explicitly and throw errors :)